GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MultithreadedReadWriteTests.cs

```diff
@@ -14,8 +14,48 @@ namespace GVFS.FunctionalTests.Tests.EnlistmentPerFixture
     [Category(Categories.Mac.M1)]
     public class MultithreadedReadWriteTests : TestsWithEnlistmentPerFixture
     {
-        [TestCase]
-        public void CanReadUnhydratedFileInParallelWithoutTearing()
+        [TestCase, Order(1)]
+        public void CanReadVirtualFileInParallel()
+        {
+            // Note: This test MUST go first, or else it needs to ensure that it is reading a unique path compared to the
+            // other tests in this class. That applies to every directory in the path, as well as the leaf file name.
+            // Otherwise, this test loses most of its value because there will be no races occurring on creating the
+            // placeholder directories, enumerating them, and then creating a placeholder file and hydrating it.
+            
+            string fileName = Path.Combine("GVFS", "GVFS.FunctionalTests", "Tests", "LongRunningEnlistment", "GitMoveRenameTests.cs");
+            string virtualPath = this.Enlistment.GetVirtualPathTo(fileName);
+
+            Exception readException = null;
+
+            Thread[] threads = new Thread[32];
+            for (int i = 0; i < threads.Length; ++i)
+            {
+                int myIndex = i;
+                threads[i] = new Thread(() =>
+                {
+                    try
+                    {
+                        FileSystemRunner.DefaultRunner.ReadAllText(virtualPath).ShouldBeNonEmpty();
+                    }
+                    catch (Exception e)
+                    {
+                        readException = e;
+                    }
+                });
+
+                threads[i].Start();
+            }
+
+            for (int i = 0; i < threads.Length; ++i)
+            {
+                threads[i].Join();
+            }
+
+            readException.ShouldBeNull("At least one of the reads failed");

```

Comment:
Can you print the Exception callstack here as well?

Reply:
It already is - see your test output up above :-)

Reply:
Should have checked the output first :) You're right I can see the exception, thanks!